src/js/components/AccordionPanel/AccordionPanel.js

```diff
@@ -1,136 +1,116 @@
-    const { dark } = theme;
-
-    setHover(dark ? 'light-4' : 'dark-3');
-
-    if (onMouseOver) onMouseOver(args);
-  };
-
-  const onHandleMouseOut = (...args) => {
-    setHover(undefined);
-
-    if (onMouseOut) onMouseOut(args);
-  };
 
-  const onHandleFocus = (...args) => {
-    const { dark } = theme;
-
-    setHover(dark ? 'light-4' : 'dark-3');
-
-    if (onFocus) onFocus(args);
-  };
-
-  const onHandleBlur = (...args) => {
-    setHover(undefined);
-
-    if (onBlur) onBlur(args);
-  };
-
-  return (
-    <AccordionContext.Consumer>
-      {panelContext => {
-        const { active, animate, onPanelChange } = panelContext;
-        const AccordionIcon = active
-          ? theme.accordion.icons.collapse
-          : theme.accordion.icons.expand;
-
-        return (
-          <Box flex={false}>
-            <Button
-              role="tab"
-              aria-selected={active}
-              aria-expanded={active}
-              onClick={onPanelChange}
-              onMouseOver={onHandleMouseOver}
-              onMouseOut={onHandleMouseOut}
-              onFocus={onHandleFocus}
-              onBlur={onHandleBlur}
-            >
-              {header || (
-                <Box align="center" direction="row" justify="between" {...rest}>
-                  {typeof label === 'string' ? (
-                    <Box pad={{ horizontal: 'xsmall' }}>
-                      <Heading
-                        level={
-                          (theme.accordion.heading &&
-                            theme.accordion.heading.level) ||
-                          4
-                        }
-                        color={hover}
-                      >
-                        {label}
-                      </Heading>
-                    </Box>
-                  ) : (
-                    label
-                  )}
-                  {AccordionIcon && (
-                    <Box pad={{ horizontal: 'small' }}>
-                      <AccordionIcon color={iconColor} />
-                    </Box>
-                  )}
+const AccordionPanel = forwardRef(
+  (
+    {
+      active,
+      animate,
+      children,
+      header,
+      label,
+      onClick,
+      onMouseOut,
+      onMouseOver,
+      onPanelChange,
+      onFocus,
+      onBlur,
+      ...rest
+    },
+    ref,
+  ) => {
+    const theme = useContext(ThemeContext);
+    const [hover, setHover] = useState(undefined);
+
+    const iconColor = useMemo(
+      () => normalizeColor(theme.accordion.icons.color || 'control', theme),
+      [theme],
+    );
+
+    const AccordionIcon = useMemo(
+      () =>
+        active ? theme.accordion.icons.collapse : theme.accordion.icons.expand,
+      [active, theme.accordion.icons],
+    );
+
+    return (
+      <Box ref={ref} flex={false} onClick={onClick}>
+        <Button
+          role="tab"
+          aria-selected={active}
+          aria-expanded={active}
+          onClick={event => {
+            console.log('!!! clicked');
```

Comment:
I think we can remove the console message.

Reply:
oops, good catch!